tests/test_eigen_tensor_avoid_stl_array.cpp

```diff
@@ -5,12 +5,10 @@
     BSD-style license that can be found in the LICENSE file.
 */
```

Comment:
All other non-`TEST_SUBMODULE`s have names that do not start with `test_`. I'm maintaining two build environments (scons, blaze ~aka bazel) that rely on that convention. It would be easier if we could rename this file to eigen_tensor_avoid_stl_array, but what does that mean for the cmake file? Would that be easier or more complicated?